authenticate-with-oauth/README.md

```diff
@@ -0,0 +1,54 @@
+# Authenticate with OAuth
+
+Authenticate with ArcGIS Online (or your own portal) using OAuth2 to access secured resources (such as private web maps or layers). Accessing secured items requires logging in to the portal that hosts them (an ArcGIS Online account, for example).
+
+![Image of authenticate with OAuth](authenticate-with-oauth.png)
+
+## Use case
+
+Your app may need to access items that are only shared with authorized users. For example, your organization may host private data layers or feature services that are only accessible to verified users. You may also need to take advantage of premium ArcGIS Online services, such as geocoding or routing services, which require a named user login.
+
+## How to use the sample
+
+When you run the sample, the app will load a web map which contains premium content. You will be challenged for an ArcGIS Online login to view the private layers. Enter a user name and password for an ArcGIS Online named user account (such as your ArcGIS for Developers account). If you authenticate successfully, the traffic layer will display, otherwise the map will contain only the public basemap layer.
+
+## Use of Model View ViewModel (MVVM)
+
+This sample takes advantage of Android's `ViewModel` to encapsulate launching the OAuth user sign in prompt, establishing the activity result contract, verifying the OAuth user's credentials.
+
+1. An Activity which launches a custom tab intent and sets its own result to the redirect url of the custom tab's result
+2. A ViewModel which launches the above activity when prompted, and when it receives the result, passes it through to the `OAuthUserSignIn` object
+3. An `ArcGISAuthenticationChallengeHandler` which prompts the ViewModel to start the sign in process with a URL if it is valid.
+
+![Image of a flowchart explaining the OAuth sample](oauth-sample-flowchart.png)
+
+## How it works
+
+1. Set the `AuthenticationManager`'s `arcGISAuthenticationChallengeHandler` to a `ArcGISAuthenticationChallengeHandler`.
+2. Create an `OAuthConfiguration` specifying the portal URL, client ID, and redirect URL.
+3. Add the OAuth configuration to the authentication manager using `OAuthUserCredential.create(oAuthConfiguration)` to get a `OAuthUserSignIn`.
+4. Set up `ViewModel` responsible for launching an OAuth user sign in prompt and retrieve the authorized redirect URI
+5. Set the URI to complete the sign in using, `OAuthUserSignIn.complete(redirectUri)`
+6. Load a map with premium content requiring authentication to automatically invoke the default authentication handler.
```

Comment:
I think this can be refined a bit: 

how about: 

1. Create an `OAuthConfiguration` specifying the portal URL, client ID, and redirect URL.
2. Create an `OAuthUserSignInActivity` that will launch the sign in page in a Custom Tab and  receive & process the redirect intent when completing the Custom Tab prompt.  
3. Set up `ViewModel` responsible for launching an OAuth user sign in prompt and retrieving the authorized redirect URI
4. Crete an `ArcGISAuthenticationChallengeHandler` that checks if the `ArcGISAuthenticationChallenge.requestUrl` matches the `OAUthUserConfiguration's`  with `canBeUserdForUrl()`. If the challenge matches, then create a `OAuthUserCredential` by Invoking the `OAuthUserSignInViewModel.promptForOAuthUserSignIn`
5. Set the `AuthenticationManager`'s `arcGISAuthenticationChallengeHandler` to the `ArcGISAuthenticationChallengeHandler` created above.
6. Load a map with premium content requiring authentication to automatically invoke the `ArcGISAuthenticationHandler`.